### PR TITLE
Updates for release v0.5.7

### PR DESCRIPTION
**Pull request by dnix101**
# About this PR
This updates adds the first release of node monitoring, where array connectivity is checked from each node in the cluster. In addition the install/upgrade configuration builder is included. This allows you to build a configuration for the initial deployment of Pure Service Orchestrator™, but can also be used to perform an upgrade of a running cluster.

## Major changes
- Add DaemonSet based node monitoring to monitor if each K8S node in the cluster can reach the appropriate IP addresses (ping) for management, iSCSI and/or NFS access.
- Add a config builder which allows you to build a <code>values.yaml</code> for the release you want to deploy.
- Add the login feature for the config builder as to not expose API tokens to unauthorised users.

## Minor changes
- Always run redis container as non-root user
- Copy application logging to <code>stdout</code> so logging is now available natively in Kubernetes
- Restructure file and directory layout in repo to be able to add the node agent (DaemonSet)
- Add latest csi-provisioner time field to show more context info for provisioned and release persistent volumes
- Improved Storage Topology visibility in Settings -> PSO view
- Minor bugfixes and enhancements

### CODE REVIEW

File: psox-app/resources/views/settings/config-builder/upgrade-upload.blade.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/upgrade-upload.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/upgrade-helper.blade.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/upgrade-helper.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/upgrade-github.blade.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/upgrade-github.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/install-upload.blade.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/install-upload.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/install-helper.blade.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/install-helper.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/install-github-version.blade.php [hunk 6]
```diff
@@ -0,0 +1,102 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/install-github-version.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/install-github-edition.blade.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/install-github-edition.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/config-download.blade.php [hunk 8]
```diff
@@ -0,0 +1,125 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/config-download.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/resources/views/settings/config-builder/config-builder.blade.php [hunk 9]
```diff
@@ -0,0 +1,247 @@
+@extends('layouts.portal')
```
Comment (pso-explorer) [inline psox-app/resources/views/settings/config-builder/config-builder.blade.php:1]: :warning: **Warning**: No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them (*Internal.NoCodeFound*).

File: psox-app/public/js/plugins/markdown/markdown-it.min.js [hunk 10]
```diff
@@ -0,0 +1 @@
...6\uDF39-\uDF3F\uDF99-\uDF9C]|\uD803[\uDF55-\uDF59]|\uD804[\uDC47-\uDC4D\uDCBB\uDCBC\uDCBE-\uDCC1\uDD40-\uDD43\uDD74\uDD75\uDDC5-\uDDC8\uDDCD\uDDDB\uDDDD-\uDDDF\uDE38-\uDE3D\uDEA9]|\uD805[\uDC4B-\uDC4F\uDC5B\uDC5D\uDCC6\uDDC1-\uDDD7\uDE41-\uDE43\uDE60-\uDE6C\uDF3C-\uDF3E]|\uD806[\uDC3B\uDE3F-\uDE46\uDE9A-\uDE9C\uDE9E-\uDEA2]|\uD807[\uDC41-\uDC45\uDC70\uDC71\uDEF7\uDEF8]|\uD809[\uDC70-\uDC74]|\uD81A[\uDE6E\uDE6F\uDEF5\uDF37-\uDF3B\uDF44]|\uD81B[\uDE97-\uDE9A]|\uD82F\uDC9F|\uD836[\uDE87-\uDE8B]|\uD83A[\uDD5E\uDD5F]/},{}],64:[function(e,r,t){r.exports=/[ \xA0\u1680\u2000-\u200A\u2028\u2029\u202F\u205F\u3000]/},{}],65:[function(e,r,t){"use strict";t.Any=e("./properties/Any/regex"),t.Cc=e("./categories/Cc/regex"),t.Cf=e("./categories/Cf/regex"),t.P=e("./categories/P/regex"),t.Z=e("./categories/Z/regex")},{"./categories/Cc/regex":61,"./categories/Cf/regex":62,"./categories/P/regex":63,"./categories/Z/regex":64,"./properties/Any/regex":66}],66:[function(e,r,t){r.exports=/[\0-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/},{}],67:[function(e,r,t){"use strict";r.exports=e("./lib/")},{"./lib/":9}]},{},[67])(67)}));
```
Comment (pso-explorer) [inline psox-app/public/js/plugins/markdown/markdown-it.min.js:1]: :warning: **Warning**: File appears to be minified and cannot be processed (*Internal.Tokenizer.Exception*).

**Review by pso-explorer — COMMENTED** [review]
**phpcs** scanning turned up:

:warning: 11 warnings 

***

**hashes-api**-scanning skipped


Posting will continue in further review(s)

File: psox-app/public/js/plugins/dropzone/dropzone.min.js [hunk 11]
```diff
@@ -0,0 +1 @@
...ndexOf(e.charAt(l++)))>>4,n=(15&r)<<4|(o=this.KEY_STR.indexOf(e.charAt(l++)))>>2,i=(3&o)<<6|(a=this.KEY_STR.indexOf(e.charAt(l++))),s.push(t),64!==o&&s.push(n),64!==a&&s.push(i),t=n=i="",r=o=a="",l<e.length;);return s}}]),e}();ExifRestore.initClass();var contentLoaded=function(t,n){function i(e){if("readystatechange"!==e.type||"complete"===o.readyState)return("load"===e.type?t:o)[s](u+e.type,i,!1),!r&&(r=!0)?n.call(t,e.type||e):void 0}var r=!1,e=!0,o=t.document,a=o.documentElement,l=o.addEventListener?"addEventListener":"attachEvent",s=o.addEventListener?"removeEventListener":"detachEvent",u=o.addEventListener?"":"on";if("complete"!==o.readyState){if(o.createEventObject&&a.doScroll){try{e=!t.frameElement}catch(e){}e&&!function t(){try{a.doScroll("left")}catch(e){return void setTimeout(t,50)}return i("poll")}()}return o[l](u+"DOMContentLoaded",i,!1),o[l](u+"readystatechange",i,!1),t[l](u+"load",i,!1)}};function __guard__(e,t){return null!=e?t(e):void 0}function __guardMethod__(e,t,n){return null!=e&&"function"==typeof e[t]?n(e,t):void 0}Dropzone._autoDiscoverFunction=function(){if(Dropzone.autoDiscover)return Dropzone.discover()},contentLoaded(window,Dropzone._autoDiscoverFunction);
```
Comment (pso-explorer) [inline psox-app/public/js/plugins/dropzone/dropzone.min.js:1]: :warning: **Warning**: File appears to be minified and cannot be processed (*Internal.Tokenizer.Exception*).

**Review by pso-explorer — COMMENTED** [review]
Previous scan continued.

File: psox-app/resources/views/settings/config-builder/upgrade-upload.blade.php (hunk 1 above)
Comment (dnix101) [reply psox-app/resources/views/settings/config-builder/upgrade-upload.blade.php:1]: Default Blade templating

File: psox-app/resources/views/settings/config-builder/upgrade-github.blade.php (hunk 3 above)
Comment (dnix101) [reply psox-app/resources/views/settings/config-builder/upgrade-github.blade.php:1]: Default Blade templating

File: psox-app/resources/views/settings/config-builder/install-helper.blade.php (hunk 5 above)
Comment (dnix101) [reply psox-app/resources/views/settings/config-builder/install-helper.blade.php:1]: Default Blade templating

File: psox-app/resources/views/settings/config-builder/install-github-edition.blade.php (hunk 7 above)
Comment (dnix101) [reply psox-app/resources/views/settings/config-builder/install-github-edition.blade.php:1]: Default Blade templating

File: psox-app/resources/views/settings/config-builder/config-builder.blade.php (hunk 9 above)
Comment (dnix101) [reply psox-app/resources/views/settings/config-builder/config-builder.blade.php:1]: Default Blade templating

**Pull request merged**